components/datetime/src/format/timezone.rs

```diff
@@ -0,0 +1,227 @@
+    time_zone_format: &TimeZoneFormat,
+    time_zone: &T,
+    w: &mut W,
+) -> Result<(), Error>
+where
+    T: TimeZoneInput,
+    W: fmt::Write + ?Sized,
+{
+    if let FieldSymbol::TimeZone(zone_symbol) = field.symbol {
+        let s = match zone_symbol {
+            fields::TimeZone::LowerZ => match u8::from(field.length) {
+                1..=3 => time_zone_format
+                    .short_specific_non_location_format(time_zone)
+                    .unwrap_or_else(|| time_zone_format.localized_gmt_format(time_zone)),
+                4 => time_zone_format
+                    .long_specific_non_location_format(time_zone)
+                    .unwrap_or_else(|| time_zone_format.localized_gmt_format(time_zone)),
+                _ => {
+                    return Err(Error::Pattern(PatternError::FieldTooLong(
+                        FieldSymbol::TimeZone(zone_symbol),
+                    )))
+                }
+            },
+            fields::TimeZone::UpperZ => match u8::from(field.length) {
+                1..=3 => time_zone_format.iso8601_format(
+                    time_zone,
+                    IsoFormat::Basic,
+                    IsoMinutes::Required,
+                    IsoSeconds::Optional,
+                ),
+                4 => time_zone_format.localized_gmt_format(time_zone),
+                5 => time_zone_format.iso8601_format(
+                    time_zone,
+                    IsoFormat::UtcExtended,
+                    IsoMinutes::Required,
+                    IsoSeconds::Optional,
+                ),
+                _ => {
+                    return Err(Error::Pattern(PatternError::FieldTooLong(
+                        FieldSymbol::TimeZone(zone_symbol),
+                    )))
+                }
+            },
+            fields::TimeZone::UpperO => match u8::from(field.length) {
+                1..=4 => time_zone_format.localized_gmt_format(time_zone),
+                _ => {
+                    return Err(Error::Pattern(PatternError::FieldTooLong(
+                        FieldSymbol::TimeZone(zone_symbol),
+                    )))
+                }
+            },
+            fields::TimeZone::LowerV => match u8::from(field.length) {
+                1 => time_zone_format
+                    .short_generic_non_location_format(time_zone)
+                    .or_else(|| time_zone_format.generic_location_format(time_zone))
+                    .unwrap_or_else(|| time_zone_format.localized_gmt_format(time_zone)),
+                4 => time_zone_format
+                    .long_generic_non_location_format(time_zone)
+                    .or_else(|| time_zone_format.generic_location_format(time_zone))
+                    .unwrap_or_else(|| time_zone_format.localized_gmt_format(time_zone)),
+                _ => {
+                    return Err(Error::Pattern(PatternError::FieldTooLong(
+                        FieldSymbol::TimeZone(zone_symbol),
+                    )))
```

Comment:
@sffc I just had the realization that `FieldTooLong` works for every case, except for this one, in which the valid lengths are just `1` and `4` with no in-between.

How do you feel about changing `FieldTooLong` to something like `FieldLengthInvalid`?